src/algorithms/synthesis/syrec_synthesis.cpp

```diff
@@ -0,0 +1,1447 @@
+#include "algorithms/synthesis/syrec_synthesis.hpp"
+
+#include "core/functions/add_gates.hpp"
+#include "core/syrec/expression.hpp"
+#include "core/syrec/program.hpp"
+#include "core/syrec/variable.hpp"
+#include "core/utils/timer.hpp"
+
+#include <boost/dynamic_bitset.hpp>
+#include <boost/format.hpp>
+#include <boost/range/adaptors.hpp>
+#include <cmath>
+#include <functional>
+#include <memory>
+#include <numeric>
+
+namespace syrec {
+    static std::stack<unsigned>               exp_opp;
+    static std::stack<std::vector<unsigned>>  exp_lhss, exp_rhss;
+    bool                                      sub_flag = false;
+    static std::vector<unsigned>              op_vec, assign_op_vector, exp_op_vector;
+    static std::vector<std::vector<unsigned>> exp_lhs_vector, exp_rhs_vector;
```

Comment:
global (static) variables are a bad design. I would argue that these should be members of the SyReC synthesizer.